packages/camera/camera_windows/windows/capture_controller.cpp

```diff
@@ -860,9 +875,31 @@ void CaptureControllerImpl::OnRecordStopped(CameraResult result,
 // Implements CaptureEngineObserver::UpdateBuffer.
 bool CaptureControllerImpl::UpdateBuffer(uint8_t* buffer,
                                          uint32_t data_length) {
+  using flutter::EncodableValue, flutter::EncodableMap;
+
   if (!texture_handler_) {
     return false;
   }
+  if (image_stream_sink_) {
+    // Encode the frame into a map that can be sent through the `EventSink`.
+    EncodableMap encoded_frame = {
+        {EncodableValue("data"),
+         EncodableValue(std::vector<uint8_t>(buffer, buffer + data_length))},
+        {EncodableValue("height"),
+         EncodableValue(static_cast<int64_t>(preview_frame_height_))},
+        {EncodableValue("width"),
+         EncodableValue(static_cast<int64_t>(preview_frame_width_))},
+        {EncodableValue("length"),
+         EncodableValue(static_cast<int64_t>(data_length))}};
+
+    task_runner_->EnqueueTask([weak_sink = std::weak_ptr(image_stream_sink_),
+                               encoded_frame = std::move(encoded_frame)]() {
+      auto sink = weak_sink.lock();
```

Comment:
The type is not trivially understandable from the RHS, so per the style guide this should not use `auto`.

Reply:
Expanded the type.